src/backend/mpi/swdsm.cpp

```diff
@@ -1010,15 +1010,15 @@ void * argo_gmalloc(unsigned long size){
 	return ptrtmp;
 }
 
-void argo_initialize(unsigned long long size){
+void argo_initialize(std::size_t argo_size, std::size_t cache_size){
 	int i;
 	unsigned long j;
 	initmpi();
 	unsigned long alignment = pagesize*CACHELINE*numtasks;
-	if((size%alignment)>0){
-		size += alignment - 1;
-		size /= alignment;
-		size *= alignment;
+	if((argo_size%alignment)>0){
+		argo_size += alignment - 1;
+		argo_size /= alignment;
+		argo_size *= alignment;
```

Comment:
For a future patch: Can this alignment rounding be placed in a function? With a good function name it would probably be clearer what kind of rounding it is. There are similar occurrences elsewhere.

Reply:
See issue #24